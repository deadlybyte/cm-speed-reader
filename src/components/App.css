@import url('https://fonts.googleapis.com/css?family=Ubuntu');

body {
  background: #18141a url(../media/images/stadium-background-x1.jpeg) no-repeat
    center center fixed;
  background-position: 50% 50%;
  background-size: cover;
  font-family: 'Ubuntu', sans-serif;
  padding: 0;
  margin: 0;
  width: 100%;
}

.CM-container {
  height: 100vh;
}

.App-header {
  background-color: #ff0000;
  line-height: 30px;
  margin: 0 0 10px;
  min-height: 40px;
}

.App-header h1 {
  color: white;
  font-size: x-large;
  letter-spacing: -1px;
  padding: 0;
  text-shadow: 3px 4px 5px #ad0000;
  word-spacing: 6px;
  word-wrap: break-word;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: xx-large;
}

.CM-overlay {
  background-color: rgba(24, 20, 16, 0.7);
  margin: 2px 0 5px;
}

.CM-overlay div {
  padding: 2px 10px;
}

.CM-overlay span {
  padding: 6px 10px;
  box-decoration-break: clone;
}

.CM-match-block {
  min-height: 80px;
}

.CM-block {
  background-color: red;
  border: 8px outset rgb(247, 76, 76);
  padding: 2px;
  text-align: center;
}

.CM-block p {
  margin: 0;
  word-wrap: break-word;
}

.CM-block span {
  color: white;
  text-shadow: 2px 1px black;
}

.CM-commentary-container {
  flex-grow: 1;
}

.CM-commentary-block {
  background-color: white;
  border: 2px outset;
  border-top-color: white;
  border-left-color: white;
  border-bottom-color: #adaaa5;
  border-right-color: #adaaa5;
  margin: 0 20px;
}

.CM-commentary-block span {
  color: #0030a5;
  text-shadow: 2px 1px #a5bade;
}

.CM-match-metadata {
  color: yellow;
}

.CM-btn {
  background-color: #210063;
  border: 1px outset;
  border-top-color: #31009c;
  border-left-color: #31009c;
  border-bottom-color: #100042;
  border-right-color: #100042;
  border-radius: 0;
  color: white;
}

.CM-btn:focus,
.CM-btn:active {
  color: white;
}

.container-fluid {
  padding-left: 0;
  padding-right: 0;
}

/* Medium devices (tablets, 768px and up) */
@media (min-width: 768px) {
  body {
    background-image: url(../media/images/stadium-background.jpeg);
  }
  .container-fluid {
    padding: 0 15px;
  }

  .App-header {
    line-height: 60px;
    margin: 10px 0;
    min-height: 80px;
  }

  .App-header h1 {
    font-size: xx-large;
  }

  .CM-block p {
    margin: 12px;
  }

  .CM-overlay {
    margin: 5px 0 10px;
  }

  .CM-overlay div {
    padding: 5px 10px;
  }
}
