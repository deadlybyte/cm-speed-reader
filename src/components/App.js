// @flow
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CMSpeedReadPlayerContainer from '../containers/CMSpeedReadPlayerContainer';
import DateDisplay from './DateDisplay';
import CommentaryControlsContainer from '../containers/CommentaryControlsContainer';
import VersionInfo from './VersionInfo';
import VenueContainer from '../containers/VenueContainer';
import WeatherContainer from '../containers/WeatherContainer';
import './App.css';
import '../media/images/stadium-background.jpeg';
import '../media/images/stadium-background-x1.jpeg';

class App extends Component {
  static defaultProps = {
    clientIp: undefined,
    fetchForecast: undefined
  };

  static propTypes = {
    clientIp: PropTypes.string,
    fetchGeoIp: PropTypes.func.isRequired,
    fetchForecast: PropTypes.func
  };

  async componentDidMount() {
    const { fetchGeoIp } = this.props;

    await fetchGeoIp();
  }

  async componentWillReceiveProps(nextProps) {
    const { clientIp } = this.props;

    if (
      nextProps.clientIp &&
      clientIp !== nextProps.clientIp &&
      nextProps.fetchForecast
    ) {
      await nextProps.fetchForecast();
    }
  }

  render() {
    return (
      <div className="container-fluid d-flex flex-column CM-container ">
        <header className="d-flex align-items-center justify-content-center App-header CM-block">
          <h1 className="App-title">Championship Manager Speed Reader</h1>
        </header>
        <div className="row no-gutters CM-overlay">
          <div className="col">
            <CommentaryControlsContainer />
          </div>
        </div>
        <div className="row no-gutters">
          <div className="col">
            <DateDisplay />
          </div>
        </div>
        <div className="row no-gutters align-items-center CM-commentary-container CM-overlay">
          <div className="col">
            <div className="CM-match-block">
              <CMSpeedReadPlayerContainer />
            </div>
          </div>
        </div>
        <div className="row no-gutters justify-items-end CM-overlay">
          <div className="col CM-match-metadata">
            <div className="row no-gutters">
              <div className="col-sm-6">
                <VersionInfo />
              </div>
              <div className="col-sm-6 ml-auto text-sm-right">
                <WeatherContainer />
              </div>
            </div>
          </div>
        </div>
        <div className="row no-gutters justify-items-end">
          <div className="col">
            <VenueContainer />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
