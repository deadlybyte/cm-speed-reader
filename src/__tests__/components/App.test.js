import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import App from '../../components/App';

describe('App', () => {
  const middlewares = [];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({
    geoIp: {
    },
    settings: {
      isPlaying: true,
      isMuted: true,
      speed: 1
    },
    weather: {
      forecast: 'Gusty, Drizzle',
      temperature: 10
    }
  });

  describe('render', () => {
    const now = Date.now;

    beforeEach(() => {
      Date.now = jest.fn(() => 1487076708000);
    });

    afterEach(() => {
      Date.now = now;
    });

    it('should render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Provider store={store}>
          <App fetchGeoIp={jest.fn} />
        </Provider>
      , div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('should render as expected', () => {
      const componentUnderTest = renderer.create(
        <Provider store={store}>
          <App fetchGeoIp={jest.fn} />
        </Provider>
      )
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });
  });

  describe('componentDidMount', () => {
    const fetchGeoIpSpy = jest.fn();

    it('should call fetchGeoIp', () => {
      shallow(
        <App fetchGeoIp={fetchGeoIpSpy} />
      );

      expect(fetchGeoIpSpy).toBeCalled();
    });
  });

  describe('componentWillRecieveProps', () => {
    let fetchForecastSpy;

    beforeEach(() => {
      fetchForecastSpy = jest.fn();
    });

    afterEach(() => {
      fetchForecastSpy.mockClear();
    });

    it('should call fetchForecast when clientIp has been set', () => {
      const componentUnderTest = shallow(
        <App fetchGeoIp={jest.fn} />
      );

      componentUnderTest.setProps({
        clientIp: '0.0.0.0',
        fetchForecast: fetchForecastSpy
      });

      expect(fetchForecastSpy).toHaveBeenCalled();
      componentUnderTest.unmount();
    });

    it('should not call fetchForecast when clientIp has been set to null', () => {
      const componentUnderTest = shallow(
        <App fetchGeoIp={jest.fn} />
      );

      componentUnderTest.setProps({
        clientIp: null,
        fetchForecast: fetchForecastSpy
      });

      expect(fetchForecastSpy).not.toHaveBeenCalled();
      componentUnderTest.unmount();
    });

    it('should not call fetchForecast when clientIp has been set to undefined', () => {
      const componentUnderTest = shallow(
        <App fetchGeoIp={jest.fn} />
      );

      componentUnderTest.setProps({
        clientIp: undefined,
        fetchForecast: fetchForecastSpy
      });

      expect(fetchForecastSpy).not.toHaveBeenCalled();
      componentUnderTest.unmount();
    });
  });
});
