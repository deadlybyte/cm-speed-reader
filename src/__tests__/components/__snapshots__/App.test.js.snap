// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render should render as expected 1`] = `
<div
  className="container-fluid"
>
  <header
    className="App-header CM-block d-flex align-items-center justify-content-center"
  >
    <h1
      className="App-title"
    >
      Championship Manager Speed Reader
    </h1>
  </header>
  <div
    className="row no-gutters"
  >
    <div
      className="col CM-overlay"
    >
      <div
        className="container"
      >
        <div
          className="row justify-content-sm-center no-gutters"
        >
          <div
            className="col-sm-auto col-xs-12"
          >
            <button
              className="CM-btn btn btn-primary btn-block"
              id="playToggle"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-fw fa-w-16 fa-pause-circle"
                title="Pause"
              />
            </button>
          </div>
          <div
            className="col-sm-auto col-xs-12"
          >
            <input
              className="form-control"
              id="commentary-speed"
              max={5}
              min={1}
              onChange={[Function]}
              title="Commentary speed - Very slow"
              type="number"
              value={1}
            />
          </div>
          <div
            className="col-sm-auto col-xs-12"
          >
            <button
              className="CM-btn btn btn-primary btn-block"
              id="muteToggle"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-fw fa-w-16 fa-volume-up"
                title="Unmute"
              />
            </button>
          </div>
          <div
            className="col-sm-auto col-xs-12"
          >
            <button
              className="CM-btn btn btn-primary btn-block"
              id="reset"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-fw fa-w-16 fa-angle-double-left"
                title="Reset"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="CM-overlay CM-date"
      >
        <span>
          Tuesday 14th February 2017
        </span>
      </div>
    </div>
  </div>
  <div
    className="row no-gutters"
  >
    <div
      className="col CM-overlay"
    >
      <div
        className="CM-match-block"
      >
        <div
          className="CM-block CM-commentary-block"
        >
          <p
            className="App-intro"
          >
            <span>
               
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row no-gutters"
  >
    <div
      className="col CM-overlay CM-match-metadata"
    >
      <div
        className="row"
      >
        <div
          className="col col-sm-4"
        >
          <span>
            Version - 
            0.1.0
          </span>
        </div>
        <div
          className="col-sm-4 ml-auto text-md-right"
        >
          <span
            className="CM-weather-forecast"
          >
            Weather - 
            Gusty, Drizzle
            <span>
              , 
              10
              °C
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="CM-venue-block"
      >
        <p>
          <span>
            Unknown
            , 
            Unknown
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
